### PR TITLE
Bail out from couple of IDE analyzers in presence of OperationKind.None operations

### LINKED ISSUE #33142: IDE0051: Private member is unused when used in an XML literal in Visual Basic
**Version Used**: 
VS 2019 Preview, Version 16.0.0 Preview 2.1

**Steps to Reproduce**:
```
    Public Class UnusedBar
        Public Sub Foo()
            Dim xml = <tag><%= Me.Bar() %></tag>
        End Sub

        Private Function Bar() As Integer
            Return 42
        End Function
    End Class
```
Bar doesn't have to be a function, it can also be a property or a field.

**Expected Behavior**:
No message is shown.

**Actual Behavior**:
Message shown: `Private member 'UnusedBar.Bar' is unused.`

### LINKED ISSUE #37988: VB: IDE0060 false positive when using XML axis properties
**Version Used**: VS 16.3 Preview 2

**Steps to Reproduce**:

1. Paste the code below into a VB file.

```vbnet
Public Class C
    Sub M(param As XElement)
        Dim a = param.<Test>
        Dim b = param.@Test
        Dim c = param...<Test>
    End Sub
End Class
```

**Expected Behavior**: `param` is not marked as unused.

**Actual Behavior**: `param` is marked as unused (IDE0060).

### PR DESCRIPTION
**Pull request by mavasani**
1. https://github.com/dotnet/roslyn/commit/49806932395686a392efea1236e2edd1b066c6f2: Bail out for unused parameters analyzer. Fixes #37988
2. https://github.com/dotnet/roslyn/commit/8a8629cbfd00c25456f43014d1061d9700ea0e0b: Bail out for unused private members analyzer. Fixes #33142

### CODE REVIEW

File: src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -152,18 +152,39 @@ private void RegisterActions(CompilationStartAnalysisContext compilationStartCon
...
                     symbolStartContext.RegisterOperationAction(AnalyzeObjectCreationOperation, OperationKind.ObjectCreation);
-                    symbolStartContext.RegisterOperationAction(_ => hasInvalidOrDynamicOperation = true, OperationKind.Invalid,
+
+                    // We bail out reporting diagnostics for named types if it contains following kind of operations:
+                    //  1. Invalid operations, i.e. erroneous code:
+                    //     We do so to ensure that we don't report false positives during editing scenarios in the IDE, where the user
+                    //     is still editing code and fixing unresolved references to symbols, such as overload resolution errors.
+                    //  2. Dynamic operations, where we do not know the exact member being referenced at compile time.
+                    //  3. Operations with OperationKind.None which are not operation root nodes. Attributes
+                    //     generate operation blocks with root operation with OperationKind.None, and we don't want to bail out for them.
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs]: is it possible we could check for that exact condition. i.e. OperationKind.None only in the attribute case?

File: src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs [hunk 2]
```diff
@@ -152,18 +152,39 @@ private void RegisterActions(CompilationStartAnalysisContext compilationStartCon
...
                     _analyzer.HandleNamedTypeSymbolStart(symbolStartContext, onSymbolUsageFound);
+
+                    return;
+
+                    void AnalyzeOperationNone(OperationAnalysisContext context)
+                    {
+                        if (context.Operation.Kind == OperationKind.None &&
+                            context.Operation.Parent != null)
+                        {
+                            hasUnsupportedOperation = true;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs]: nit: `hasUnsupportedOperation |= context.Operation.Kind == OperationKind.None && context.Operation.Parent != null`

File: src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs [hunk 3]
```diff
@@ -322,7 +322,7 @@ private static bool IsHandledLocalOrParameterReferenceTreeShape(IOperation opera
                 /// Method invoked in <see cref="AnalyzeOperationBlockEnd(OperationBlockAnalysisContext)"/>
                 /// for each operation block to determine if we should analyze the operation block or bail out.
                 /// </summary>
-                private bool ShouldAnalyze(IOperation operationBlock, ISymbol owningSymbol)
+                private bool ShouldAnalyze(IOperation operationBlock, ISymbol owningSymbol, ref bool hasOperationNoneDescendant)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs]: nit: would prefer either all return types (i.e. tuple) or all out-params.  (just a nit).

File: src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs (hunk 1 above)
Comment (mavasani) [reply src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs]: Unfortunately, not without going back to syntax.

File: src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs (hunk 2 above)
Comment (mavasani) [reply src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs]: I had used that first, but just felt this was bit more readable.

**Comment by mavasani** [comment]
@dotnet/roslyn-ide Can I get a quick review? Trivial bail out change from couple of IDE analyzers to avoid false positives.

**Pull request merged**